src/components/input/_macro.njk

```diff
@@ -16,29 +16,47 @@
-            {% if params.autocomplete is defined %}autocomplete="{{ params.autocomplete }}"{% endif %}
-            {% if params.charcheck %}data-char-check-ref="{{ params.id }}-check-remaining" data-char-check-num="{{ params.charcheck }}"{% endif %}
-            {% if params.attributes %}{% for attribute, value in (params.attributes.items() if params.attributes is mapping and params.attributes.items else params.attributes) %}{{ attribute }}{% if value %}="{{ value }}"{% endif %} {% endfor %}{% endif %}
-        />
-    {% endset %}
+        {% if params.typeaheadData %}
+            <div
+                id="{{ params.id }}-container"
+                class="typeahead js-typeahead js-typeahead-input typeahead__input {% if params.classes %} {{ params.classes }}{% endif %}"
+                data-content="{{ params.content | dump }}"
+                typeahead-data="{{ params.typeaheadData }}"
+            >
+        {% endif %}
+
+                <input
+                    type="{{ type }}"
+                    id="{{ params.id }}"
+                    class="input input--text input-type__input typeahead__input js-typeahead-input {{ params.classes }}{{ exclusiveClass }}"
+                    {% if params.prefix or params.suffix %}title="{{ params.prefix.title if params.prefix }}{{ params.suffix.title if params.suffix }}"{% endif %}
+                    {% if params.name is defined %}name="{{ params.name }}"{% endif %}
+                    {% if params.accessiblePlaceholder is defined %}required{% endif %}
+                    {% if params.value is defined %}value="{{ params.value }}"{% endif %}
+                    {% if params.accept is defined %}accept="{{ params.accept }}"{% endif %}
+                    {% if params.min is defined %}min="{{ params.min }}"{% endif %}
+                    {% if params.max is defined %}max="{{ params.max }}"{% endif %}
+                    {% if params.minLength is defined %}minlength="{{ params.minLength }}"{% endif %}
+                    {% if params.maxLength is defined %}maxlength="{{ params.maxLength }}"{% endif %}
+                    {% if pattern is defined %}pattern="{{ pattern }}"{% endif %}
+                    {% if inputmode is defined %}inputmode="{{ inputmode }}"{% endif %}
+                    {% if params.autocomplete is defined %}autocomplete="{{ params.autocomplete }}"{% endif %}
+                    {% if params.charcheck %}data-char-check-ref="{{ params.id }}-check-remaining" data-char-check-num="{{ params.charcheck }}"{% endif %}
+                    {% if params.attributes %}{% for attribute, value in (params.attributes.items() if params.attributes is mapping and params.attributes.items else params.attributes) %}{{ attribute }}{% if value %}="{{ value }}"{% endif %} {% endfor %}{% endif %}
+                />
 
+        {% if params.typeaheadData %}
+                <div class="typeahead__results js-typeahead-results">
```

Comment:
The results aren't a child of `typeahead` (Unless I'm not seeing a div). So this should just be `typeahead-input-results`.

I think this sort of thing makes the case for taking some of the scss for typeahead out of the `input.scss` file. If you're not directly styling the `input` anything in the wrapper or results should probably go in a `typeahead_ui.scss` file alongside `input.scss`